### PR TITLE
admin token cache

### PR DESCRIPTION
**Pull request by jesuspg**
@hmunfru @jframos

### CODE REVIEW

**Comment by hmunfru** [comment]
LGTM

File: rest-api/src/main/java/com/telefonica/euro_iaas/paasmanager/rest/util/AdminTokenCache.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import java.io.InputStream;
+
+import net.sf.ehcache.Cache;
+import net.sf.ehcache.CacheManager;
+import net.sf.ehcache.Element;
+import net.sf.ehcache.config.CacheConfiguration;
+
+import org.openstack.docs.identity.api.v2.AuthenticateResponse;
+
+public class AdminTokenCache {
```
Comment (flopezag) [inline rest-api/src/main/java/com/telefonica/euro_iaas/paasmanager/rest/util/AdminTokenCache.java]: provide class javadoc and different public methods javadoc.

File: rest-api/src/main/java/com/telefonica/euro_iaas/paasmanager/rest/util/AdminTokenCache.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        try {
+            InputStream inputStream = this.getClass().getResourceAsStream("/ehcache.xml");
+            singletonManager = CacheManager.newInstance(inputStream);
+        } catch (Exception e) {
+            singletonManager = CacheManager.create();
+            singletonManager.addCache(CACHE_NAME);
+            cache.getCacheConfiguration();
+            cache = singletonManager.getCache(CACHE_NAME);
+            CacheConfiguration cacheConfiguration = cache.getCacheConfiguration();
+            cacheConfiguration.setTimeToIdleSeconds(1200);
```
Comment (flopezag) [inline rest-api/src/main/java/com/telefonica/euro_iaas/paasmanager/rest/util/AdminTokenCache.java]: Magic number

File: rest-api/src/main/java/com/telefonica/euro_iaas/paasmanager/rest/auth/OpenStackAuthenticationProvider.java [hunk 3]
```diff
@@ -171,39 +176,36 @@ public PaasManagerUser authenticationFiware(String token, String tenantId) {
...
-                                    .header("X-Auth-Token", newCredentials[0]).get(AuthenticateResponse.class));
-
-                } else if ((response.getStatus() == CODE_403) || (response.getStatus() == CODE_404)) {
-                    throw new BadCredentialsException("Token not valid");
+                WebTarget tokens = webResource.path("tokens").path(token);
+                Invocation.Builder builder = tokens.request();
+                response = builder.accept(MediaType.APPLICATION_XML).header("X-Auth-Token", credential[0]).get();
+                authenticateResponse = response.readEntity(AuthenticateResponse.class);
+
+                if (response.getStatus() == 200) {
```
Comment (flopezag) [inline rest-api/src/main/java/com/telefonica/euro_iaas/paasmanager/rest/auth/OpenStackAuthenticationProvider.java]: CODE_200?

**Comment by flopezag** [comment]
LGTM

**Pull request merged**